subprojects/docs/src/docs/userguide/publishing_maven.adoc

```diff
@@ -106,6 +106,23 @@ include::sample[dir="signing/maven-publish/groovy",files="build.gradle[tags=pom-
 include::sample[dir="signing/maven-publish/kotlin",files="build.gradle.kts[tags=pom-customization]"]
 ====
 
+[[publishing_maven:resolved_dependencies]]
+=== Customizing dependencies versions
+
+The set of dependencies and constraints that are added to the POM file will use by default the _declared_ versions from the build file itself.
+Sometimes it is desirable to publish the _resolved_ version instead.
+
+This is done by using the `versionMapping` DSL method which allows to configure the link:{javadocPath}/org/gradle/api/publish/VersionMappingStrategy.html[VersionMappingStrategy]:
+
+.Using resolved versions
+====
+include::sample[dir="signing/maven-publish/groovy",files="build.gradle[tags=versions-resolved]"]
+include::sample[dir="signing/maven-publish/kotlin",files="build.gradle.kts[tags=versions-resolved]"]
+====
+
+In this example above, Gradle will use the versions resolved on the `runtimeClasspath` for dependencies declared in `api` and mapped to Maven `compile`.
+For the Maven `runtime` dependencies, that come from `implementation`, Gradle will use the default resolved versions, which are coming from `runtimeClasspath` again.
```

Comment:
I think this needs some more clarification. Maybe replace with:

-> _Gradle will also use the versions resolved on the `runtimeClasspath` for dependencies declared in `implementation`, which are mapped to Maven's runtime scope. This is, because `fromResolutionResult()` indicates that Gradle should use the default classpath of a variant and  `runtimeClasspath` is the default classpath of `java-api`_.